compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -299,14 +300,31 @@ private void replaceMockedFunctions(BIRPackage birPkg, Map<String, String> mockF
                 if (bbTerminator.kind.equals(InstructionKind.CALL)) {
                     //We get the callee and the name and generate 'calleepackage#name'
                     BIRTerminator.Call callTerminator = (BIRTerminator.Call) bbTerminator;
-                    String functionKey = callTerminator.calleePkg.toString() + MOCK_ANNOTATION_DELIMITER
-                            + callTerminator.name.toString();
-                    if (mockFunctionMap.get(functionKey) != null) {
-                        // Just "get" the reference. If this doesnt work then it doesnt exist
-                        String mockfunctionName = mockFunctionMap.get(functionKey);
-                        callTerminator.name = new Name(mockfunctionName);
-                        callTerminator.calleePkg = function.pos.src.pkgID;
+
+                    // This will be an object function
+                    if (callTerminator.isVirtual && callTerminator.args.size() == 1) {
+                        String objectPkg = callTerminator.args.get(0).variableDcl.type.toString();
+                        String objectKey = objectPkg + MOCK_ANNOTATION_DELIMITER + callTerminator.name.toString();
```

Comment:
The generated object key will look something like this : 

`org/<module_name>:<version>:<objectName>#<functionName>`.

This is the same format as the key added to the `mockFunctionMap` in the testAnnotationProcessor class